x/stakeibc/keeper/redeem_stake.go

```diff
@@ -0,0 +1,156 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v22/utils"
+	recordstypes "github.com/Stride-Labs/stride/v22/x/records/types"
+	"github.com/Stride-Labs/stride/v22/x/stakeibc/types"
+)
+
+func (k Keeper) RedeemStake(ctx sdk.Context, msg *types.MsgRedeemStake) (*types.MsgRedeemStakeResponse, error) {
```

Comment:
Note: I didn't diff this, but assuming it's a direct copy paste from the msg_server.

Reply:
yeah [this](https://github.com/Stride-Labs/stride/pull/1215/commits/f8148e63a262d486f2b97c9a359e1dab9f626a52) commit was just a copy paste